lib/Util.cpp

```
@@ -0,0 +1,73 @@
+/*
+  Helper class
+
+  Copyright (c) 2016 - 2020 SUSE LLC
+
+  This program is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "Log.h"
+#include "Util.h"
+#include "Exceptions.h"
+#include <algorithm>
+using namespace std;
+
+string Util::exec(const string cmd) {
+    array<char, 128> buffer;
+    string result;
+
+    tulog.info("Executing `", cmd, "`:");
+
+    auto pipe = popen(cmd.c_str(), "r");
+
+    if (!pipe)
+        throw runtime_error{"popen() failed!"};
+
+    while (!feof(pipe)) {
+        if (fgets(buffer.data(), 128, pipe) != nullptr)
+            result += buffer.data();
+    }
+
+    int rc = pclose(pipe);
+
+    if (rc == EXIT_SUCCESS) {
+        tulog.info("◸", result, "◿");
+    } else {
+        throw ExecutionException{"`" + cmd + "` returned with error code " + to_string(rc%255) + ".", rc};
+    }
+
+    return result;
+}
+
+// trim from start (in place)
+void Util::ltrim(string &s) {
+    s.erase(s.begin(), std::find_if(s.begin(), s.end(),
+            std::not1(std::ptr_fun<int, int>(std::isspace))));
+}
+
+// trim from end (in place)
+void Util::rtrim(string &s) {
+    s.erase(std::find_if(s.rbegin(), s.rend(),
+            std::not1(std::ptr_fun<int, int>(std::isspace))).base(), s.end());
+}
+
+void Util::stub(string option) {
```

Unused, should be removed. If not, should throw an exception instead to inform the callers.